lib/src/dart_formatter.dart

```diff
@@ -16,11 +16,21 @@ import 'package:analyzer/src/string_source.dart';
 import 'package:pub_semver/pub_semver.dart';
 
 import 'exceptions.dart';
+import 'front_end/ast_node_visitor.dart';
 import 'source_code.dart';
 import 'source_visitor.dart';
 import 'string_compare.dart' as string_compare;
 import 'style_fix.dart';
 
+/// The in-progress "tall" formatting style is enabled by passing an experiment
+/// flag with this name.
+///
+/// Note that this isn't a real Dart SDK experiment: Only the formatter supports
+/// it. We use the [experimentFlags] API to pass this in so that support for it
+/// can be removed in a later version without it being a breaking change to the
+/// dart_style library API.
+const _tallStyleExperimentFlag = 'tall-style';
```

Comment:
Should this be public instead of repeated in other files?

Reply:
dart_formatter.dart is exported by the package and I don't want the flag to be part of the public API (expect as a string you can pass in) so that I can remove it without it being a breaking change.

Moved it to constants.dart and made it public there.